pepper-apis/src/main/java/org/broadinstitute/ddp/copy/AnswerToAnswerCopier.java

```diff
@@ -0,0 +1,173 @@
+import org.broadinstitute.ddp.db.dto.QuestionDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.instance.FormResponse;
+import org.broadinstitute.ddp.model.activity.instance.answer.AgreementAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.model.activity.instance.answer.AnswerRow;
+import org.broadinstitute.ddp.model.activity.instance.answer.BoolAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.CompositeAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.DateAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.DateValue;
+import org.broadinstitute.ddp.model.activity.instance.answer.NumericIntegerAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.PicklistAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.SelectedPicklistOption;
+import org.broadinstitute.ddp.model.activity.instance.answer.TextAnswer;
+import org.broadinstitute.ddp.model.activity.types.QuestionType;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AnswerToAnswerCopier {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AnswerToAnswerCopier.class);
+
+    private final Handle handle;
+    private final long operatorId;
+    private final JdbiCompositeQuestion jdbiCompositeQuestion;
+    private final JdbiCompositeAnswer jdbiCompositeAnswer;
+    private final AnswerDao answerDao;
+    private final Map<Long, CompositeQuestionDto> parentDtosByChildId = new HashMap<>();
+
+    public AnswerToAnswerCopier(Handle handle, long operatorIdForNewAnswers) {
+        this.handle = handle;
+        this.operatorId = operatorIdForNewAnswers;
+        this.jdbiCompositeQuestion = handle.attach(JdbiCompositeQuestion.class);
+        this.jdbiCompositeAnswer = handle.attach(JdbiCompositeAnswer.class);
+        this.answerDao = handle.attach(AnswerDao.class);
+    }
+
+    public void copy(FormResponse sourceInstance, QuestionDto sourceQuestion,
+                     FormResponse targetInstance, QuestionDto targetQuestion) {
+        Answer sourceAnswer = sourceInstance.getAnswer(sourceQuestion.getStableId());
+        if (sourceAnswer != null) {
+            copySourceAnswer(sourceAnswer, targetInstance, targetQuestion);
+        } else {
+            Answer parentAnswer = retrieveParentQuestion(sourceQuestion)
+                    .map(parentDto -> (CompositeAnswer) sourceInstance.getAnswer(parentDto.getStableId()))
+                    .orElse(null);
+            if (parentAnswer != null) {
+                copyFromParentAnswer((CompositeAnswer) parentAnswer, sourceQuestion, targetInstance, targetQuestion);
+            } else {
+                LOG.info("No source answer to copy from activity instance {} and question {}",
+                        sourceInstance.getGuid(), sourceQuestion.getStableId());
```

Comment:
Are we precluding copying to/from a simple question/composite question?
Very realistic case is composite question with allowMultiple=false. E.g., a composite question/multiple=false with children "firstName" and "lastName". In this case each child ought to be treated a just a "simple" question in the activity instance and copied to/from any other.

Reply:
_Re: method doing too much._ Out-sourcing the answer lookup to the caller is definitely an alternative. The caller (currently only `CopyExecutor`) has all the question DTOs too, so might be able to leverage some sharing of parent question DTOS between the copiers.

_Re: precluding._ Let's see if I understand the question. Are you asking if it allows copying from a child question to a non-composite question? This is not supported right now because I have not implemented it. The use-case is definitely reasonable, but the ticket is already a big piece of work and it doesn't seem like we will leverage that feature any time soon, so I took a shortcut here. I should have listed it as another limitation in `studies/study-schema.conf`, sorry about that. It will be a nice exercise for folks who want to jump in on the copier code!